cmake/kokkos_functions.cmake

```diff
@@ -848,6 +851,26 @@ function(COMPILER_SPECIFIC_OPTIONS_HELPER)
     endif()
   endif()
 
+  set(CMAKE_REQUIRED_QUIET ON)
+  if(PARSE_COMPILE_OPTIONS AND COMPILER_SPECIFIC_FLAGS_TMP)
+    foreach(flag IN LISTS COMPILER_SPECIFIC_FLAGS_TMP)
```

Comment:
Flags are tested one by one; perhaps it is possible to have some flags that work by pair or on the opposite, some conflicting flags.

What about testing all the flags at once, and if it does not work loop to find the fault?

Reply:
I like the idea. The CMake function allows testing multiple flags (but the passed flags need to be a list with whitespace ... we have one with semicolon ... but that can be arranged).
Another problem is if we want to trace a flag back to the `Kokkos_ENABLE_` option that added it. We would either need to sprinkle the checks all over the place or have it (like here at the moment) in one function and pass the string of the option that added the flag to `COMPILER_SPECIFIC_OPTIONS_HELPER`.

Reply:
It tests sets of flags we currently add now [bd7ba](https://github.com/kokkos/kokkos/pull/7891/commits/fe27d51117caf7987c4c2ecfc4a84cb9146bd7ba).

This should allow us to check flags that require to be paired. But we now have limited traceability back to the options that introduced them. See discussion above